detections/endpoint/dns_exfiltration_using_nslookup_app.yml

```diff
@@ -0,0 +1,56 @@
+name: DNS Exfiltration Using Nslookup App
+id: 2452e632-9e0d-11eb-bacd-acde48001122
+version: 1
+date: '2021-04-15'
+author: Teoderick Contreras, Splunk
+type: batch
+datamodel:
+- Endpoint
+description: this search is to detect potential DNS exfiltration using nslookup application.
+  This technique are seen in couple of malware and APT group to exfiltrated collected
+  data in a infected machine or infected network. This detection is looking for unique
+  use of nslookup where it tries to use specific record type, TXT, A, AAAA, that
+  are commonly used by attacker and also the retry parameter which is designed to
+  query C2 DNS multiple tries.
+search: '| tstats `security_content_summariesonly` values(Processes.process) as cmdline
+  values(Processes.process_id) as nslookup_procid values(Processes.parent_process)
+  as parent_cmdline count min(_time) as firstTime max(_time) as lastTime from datamodel=Endpoint.Processes
+  where Processes.process_name = "nslookup.exe" Processes.process = "*-querytype=*"
+  OR Processes.process="*-qt=*" OR Processes.process="*-q=*" OR Processes.process="-type=*"
+  OR Processes.process="*-retry=*" by Processes.dest Processes.user Processes.parent_process
+  Processes.process_name Processes.process Processes.process_id | `drop_dm_object_name(Processes)`
```

Comment:
Another small thing, when we write a tstats search : we should either do values(fields) in the beginning of the search and exclude those fields in the by clause or vice versa,  since in this case by the looks of it will have two similar values in different field name:

Eg: process_id and nslookup_procid will have similar values.

Reply:
thanks for this info. will fix this also :)